### PR TITLE
Fix a bunch more compilation warnings

### PR DESCRIPTION
**Pull request by shellixyz**

### CODE REVIEW

File: src/main/common/filter.c [hunk 1]
```diff
@@ -172,16 +174,23 @@ void biquadFilterInit(biquadFilter_t *filter, uint16_t filterFreq, uint32_t samp
     }
     else {
         // Not possible to filter frequencies above Nyquist frequency - passthrough
-        filter->b0 = 1.0f;
-        filter->b1 = 0.0f;
-        filter->b2 = 0.0f;
-        filter->a1 = 0.0f;
-        filter->a2 = 0.0f;
+        goto initError;
```
Comment (digitalentity) [inline src/main/common/filter.c]: Why goto? It's generally considered non-clean programming style, hurts understanding of the code which can lead to hard to catch bugs.

Comment (shellixyz) [reply src/main/common/filter.c]: Seems to me that it is the easiest solution avoiding code duplication. Error handling is often done like this. What would you prefer ? I can make a function to initialize the filter as passthrough or perhaps just make the default case not do anything

Comment (digitalentity) [reply src/main/common/filter.c]: I think we should initialize the filter as passthrough. Not doing anything is not clean either.

Comment (shellixyz) [reply src/main/common/filter.c]: What about this new solution ? Doesn't seem any better to me than using goto though.

Comment (digitalentity) [reply src/main/common/filter.c]: It's definitely more readable now (at least to me). Thanks!

**Pull request merged**